File: src/record.cpp

#include "Record.hpp"

#include <algorithm>
#include <iostream>

using namespace std;

void Record::addElement(string data) {
    this->data.push_back(data);
}

string Record::getElement(unsigned int position) {
    return this->data.at(position);
}

void Record::print(){
    for_each(data.begin(), data.end(), [](string& e){
            cout<<e<<"|";    
        });
    cout<<endl;
}

File: src/Record.java
import java.util.Vector;

public class Record {
    public Vector<String> data;

    public Record(int size) {
        this.data = new Vector<String>(size);
    }

    public void print() {
        for (int i=0; i<data.size(); i++)
            System.out.print(data.get(i)+" ");
        System.out.println();
    }
}
